scripts/release/audit_rosdistro.py

```diff
@@ -0,0 +1,130 @@
+
+
+def run_audit(config_url, rosdistro_name, cache_dir):
+    config = get_config_index(config_url)
+    index = get_index(config.rosdistro_index_url)
+    dist_file = get_distribution_file(index, rosdistro_name)
+    dist_cache = get_distribution_cache(index, rosdistro_name)
+    build_files = get_release_build_files(config, rosdistro_name)
+    missing_packages = {}
+    for bf_name, bf_value in build_files.items():
+        missing_packages[bf_name] = copy.deepcopy(bf_value.targets)
+        for target in bf_value.get_targets_list():
+            all_pkgs, missing_pkgs = partition_packages(
+                config_url, rosdistro_name, bf_name, target, cache_dir,
+                deduplicate_dependencies=True, dist_cache=dist_cache)
+            missing_packages[bf_name][target] = missing_pkgs
+            if 'all' in missing_packages[bf_name]:
+                missing_packages[bf_name]['all'] &= missing_pkgs
+            else:
+                missing_packages[bf_name]['all'] = missing_pkgs
+
+            if 'all' in missing_packages:
+                missing_packages['all'] &= missing_pkgs
+            else:
+                missing_packages['all'] = missing_pkgs
+
+    recommended_actions = len(missing_packages['all'])
+    print("# Sync preparation report for %s " % rosdistro_name)
+    print("Prepared for configuration: %s" % config_url)
+    print("Prepared for rosdistro index: %s" % config.rosdistro_index_url)
+
+    print("## Packages failing on all platforms\n Release should be rolled back:")
+    for mp in sorted(missing_packages['all']):
+        print(" - %s " % mp)
+
+    def get_package_repository_link(dist_file, pkg_name):
+        """Return the best guess of the url for filing a ticket against the package"""
+        pkg = dist_file.release_packages[pkg_name]
+        repo_name = pkg.repository_name
+        repo = dist_file.repositories[repo_name]
+        if repo.source_repository and repo.source_repository.url:
+            return repo.source_repository.url
+        if repo.release_repository and repo.release_repository.url:
+            return repo.release_repository.url
+        return None
+
+    for bf_name in build_files.keys():
+        print("## Packages failing for buildfile %s\n\nPackages should be blacklisted:" % bf_name)
+        # TODO(tfoote) use rosdistro API to print the release build config for editing
+        recommended_blacklists = sorted(missing_packages[bf_name]['all'] - missing_packages['all'])
+        recommended_actions += len(recommended_blacklists)
+        for rb in recommended_blacklists:
+            print(' - %s:' % rb)
+            jenkins_urls = get_jenkins_job_urls(
+                rosdistro_name,
+                config.jenkins_url,
+                bf_name,
+                build_files[bf_name].get_targets_list())
+            url = get_package_repository_link(dist_file, rb)
+            print(f'   - Suggested ticket location [{url}]({url})')
```

Comment:
While the script has a Python 3 shebang all scripts in this package currently support Python 2.7 as well as minor version of Python 3 which don't support f-strings. So please use normal string instead.

That being said it would be good to add an invocation of this new script to the CI file to ensure it generates the output as desired.